sdk/python/kfp/dsl/_pipeline_param.py

```diff
@@ -249,3 +249,14 @@ def ignore_type(self):
     """ignore_type ignores the type information such that type checking would also pass"""
     self.param_type = None
     return self
+
+
+def additional_input_name_for_pipelineparam(
```

Comment:
This function seems specific to the IR ComponentSpec conversion. Perhaps move this to sdk/python/kfp/dsl/component_spec.py